src/Filter/FieldDefinitionAdapter/CalculatedValue.php

```diff
@@ -28,7 +28,7 @@ class CalculatedValue extends DefaultAdapter implements FieldDefinitionAdapterIn
     protected function doGetIndexDataValue($object, $ignoreInheritance = false)
     {
         $name = $this->fieldDefinition->getName();
-        $value = $object->getValueForFieldName($name);
+        $value = $this->loadRawDataFromContainer($name);
```

Comment:
`loadRawDataFromContainer` calls field getter, which shouldn't be the case for calculated fields as it needs to `Service::getCalculatedFieldValue` to get the value based on calculator class.

What exactly is the problem that you are trying to solve?

Reply:
Hi, this change was hinted by @fashxp 
When adding a calculated value at field collection level, we get the error…
`
Message: Call to undefined method getValueForFieldName in class Pimcore\Model\DataObject\Fieldcollection\Data\CampaignCreative Trace: #0 /app/vendor/pimcore/advanced-object-search/src/Filter/FieldDefinitionAdapter/CalculatedValue.php(31): Pimcore\Model\AbstractModel->__call('getValueForFiel...', Array)
`
The method getValueForFieldName is part of the Concrete class and not AbstractData.
I think this happens when the calculated value is within a fieldcollection and not part of a data object directly

Reply:
ok, it should work as field getter also call `Service::getCalculatedFieldValue`. please pass object container then:
```suggestion
        $value = $this->loadRawDataFromContainer($object, $name);
```

Reply:
I adjusted the PR here https://github.com/pimcore/advanced-object-search/pull/130 with a different user